src/single-spa-react.js

```diff
@@ -46,7 +46,9 @@ function bootstrap(opts) {
 
 function mount(opts, props) {
   return new Promise((resolve, reject) => {
-    const whenFinished = resolve;
+    const whenFinished = function() {
```

Comment:
Yes this needs to be a normal function and can not be an arrow function. The reason is that react calls the callback with a bound context. See: https://github.com/facebook/react/blob/37e4329bc81def4695211d6e3795a654ef4d84f5/packages/react-reconciler/src/ReactFiberUpdateQueue.js#L352